extension/json/src/functions/table_functions/json_scan.cpp

```diff
@@ -0,0 +1,502 @@
+    prevBufferRemainder = 0;
+    memset(bufferPtr + bufferSize, 0, YYJSON_PADDING_SIZE);
+    return true;
+}
+
+uint64_t JSONScanLocalState::readNext() {
+    numValuesToOutput = 0;
+    while (numValuesToOutput == 0) {
+        if (bufferOffset == bufferSize) {
+            if (!readNextBuffer()) {
+                break;
+            }
+        }
+        parseNextChunk();
+    }
+    return numValuesToOutput;
+}
+
+struct JsonScanBindData : public ScanBindData {
+    case_insensitive_map_t<uint64_t> colNameToIdx;
+    bool scanFromList;
+    bool scanFromStruct;
+    JsonScanFormat format;
+    uint64_t numRows;
+
+    JsonScanBindData(std::vector<common::LogicalType> columnTypes,
+        std::vector<std::string> columnNames, ReaderConfig config, main::ClientContext* ctx,
+        case_insensitive_map_t<uint64_t> colNameToIdx, bool scanFromList, bool scanFromStruct,
+        JsonScanFormat format, uint64_t numRows)
+        : ScanBindData(std::move(columnTypes), std::move(columnNames), std::move(config), ctx),
+          colNameToIdx{std::move(colNameToIdx)}, scanFromList{scanFromList},
+          scanFromStruct{scanFromStruct}, format{format}, numRows{numRows} {}
+
+    uint64_t getFieldIdx(const std::string& fieldName) const;
+
+    std::unique_ptr<TableFuncBindData> copy() const override {
+        return std::unique_ptr<JsonScanBindData>(new JsonScanBindData(*this));
+    }
+
+private:
+    JsonScanBindData(const JsonScanBindData& other)
+        : ScanBindData{other}, colNameToIdx{other.colNameToIdx}, scanFromList{other.scanFromList},
+          scanFromStruct{other.scanFromStruct}, format{other.format}, numRows{other.numRows} {}
+};
+
+uint64_t JsonScanBindData::getFieldIdx(const std::string& fieldName) const {
+    auto normalizedName = fieldName;
+    // TODO(Ziyi): this is a temporary fix for the json testing framework, since copy-to-json
+    // always outputs the property name, so we have to remove the property name prefix for
+    // matching.
+    if (!colNameToIdx.contains(fieldName)) {
+        std::regex pattern(R"(^[^.]+\.(.*))");
+        std::smatch match;
+        if (std::regex_match(fieldName, match, pattern) && match.size() > 1) {
+            normalizedName = match[1]; // Return the part of the string after the "xxx."
+        }
+    }
+    return colNameToIdx.contains(normalizedName) ? colNameToIdx.at(normalizedName) : UINT64_MAX;
+}
+
+static std::unique_ptr<TableFuncBindData> bindFunc(main::ClientContext* context,
+    ScanTableFuncBindInput* scanInput) {
+    std::vector<LogicalType> columnTypes;
+    std::vector<std::string> columnNames;
+    JsonScanConfig scanConfig(scanInput->config.options);
+    auto parsedJson = fileToJson(context, scanInput->inputs[0].strVal, scanConfig.format);
+    auto scanFromList = false;
+    auto scanFromStruct = false;
+    if (scanInput->expectedColumnNames.size() > 0) {
+        columnTypes = LogicalType::copy(scanInput->expectedColumnTypes);
```

Comment:
I think `copy` works here. And we should slowly remove `LogicalType::copy` from the codebase.